clean/acadia_so_cprr.py

```diff
@@ -0,0 +1,52 @@
+import sys
+sys.path.append('../')
+from lib.path import data_file_path
+import pandas as pd
+from lib.columns import clean_column_names, set_values
+from lib.clean import clean_dates
+from lib.uid import gen_uid
+
+
+def clean_names(df):
+    names = df.deputy.str.lower().str.strip().fillna('')\
+        .str.extract(r'(\w+) ?(\w+)? (\w+)')
+    
+    df.loc[:, 'first_name'] = names[0].fillna('')
+    df.loc[:, 'middle_name'] = names[1].fillna('')
+    df.loc[:, 'last_name'] = names[2].fillna('')
+    return df[~((df.first_name ==  '') & (df.last_name ==  ''))].drop(columns='deputy')
+
+
+def clean_allegation(df):
+    df.loc[:, 'allegation'] = df.allegation.str.lower().str.strip()\
+        .str.replace(r'\bia\b', 'internal affairs', regex=True)
+    return df 
+
+
+def clean_action(df):
+    df.loc[:, 'action'] = df.action.str.lower().str.strip()\
+        .str.replace(r'^(\w{1}) day', r'\1-day', regex=True)
+    return df 
+
+
+def clean():
+    df = pd.read_csv(data_file_path('raw/acadia_so/cprr_acadia_so_2018_2021.csv'))\
+        .pipe(clean_column_names)\
+        .rename(columns={
+            'date': 'receive_date'
+        })\
+        .pipe(clean_dates, ['receive_date'])\
+        .pipe(clean_names)\
```

Comment:
Are the names in lower case and with punctuations removed? Even if they are I think it's still good practice to always use the `lib.clean.clean_names` function on the name columns no matter what. That way the names during matching are always uniform.

Reply:
Good call. Thank you.